vpnglean/ios.cmake

```diff
@@ -0,0 +1,118 @@
+    ${GLEAN_VENDORED_PATH}/glean-core/ios/Glean/Net/HttpPingUploader.swift
+    ${GLEAN_VENDORED_PATH}/glean-core/ios/Glean/Scheduler/GleanLifecycleObserver.swift
+    ${GLEAN_VENDORED_PATH}/glean-core/ios/Glean/Scheduler/MetricsPingScheduler.swift
+    ${GLEAN_VENDORED_PATH}/glean-core/ios/Glean/Utils/Logger.swift
+    ${GLEAN_VENDORED_PATH}/glean-core/ios/Glean/Utils/Sysctl.swift
+    ${GLEAN_VENDORED_PATH}/glean-core/ios/Glean/Utils/Unreachable.swift
+    ${GLEAN_VENDORED_PATH}/glean-core/ios/Glean/Utils/Utils.swift
+)
+target_sources(iosglean PUBLIC
+    ${GLEAN_VENDORED_PATH}/glean-core/ios/Glean/Glean.h
+)
+
+# Build gleanFFI.h and glean.swift files using UniFFI
+# and build the internal Glean metrics file
+execute_process(
+    COMMAND 
+        ${CARGO_EXECUTABLE} run --manifest-path ${GLEAN_VENDORED_PATH}/tools/embedded-uniffi-bindgen/Cargo.toml 
+            -- generate -l swift -o ${CMAKE_CURRENT_BINARY_DIR}/glean
+            ${GLEAN_VENDORED_PATH}/glean-core/src/glean.udl
+    COMMAND ${CMAKE_COMMAND} -E env     
+            SOURCE_ROOT=${CMAKE_CURRENT_BINARY_DIR} 
+            PROJECT=${BUILD_IOS_APP_IDENTIFIER} 
+        ${GLEAN_VENDORED_PATH}/glean-core/ios/sdk_generator.sh 
+            -o ${CMAKE_CURRENT_BINARY_DIR}/glean/generated --allow-reserved
+            ${GLEAN_VENDORED_PATH}/glean-core/metrics.yaml ${GLEAN_VENDORED_PATH}/glean-core/pings.yaml
+    COMMAND_ERROR_IS_FATAL ANY
+)
+target_sources(iosglean PRIVATE 
+    ${CMAKE_CURRENT_BINARY_DIR}/glean/gleanFFI.modulemap
+    ${CMAKE_CURRENT_BINARY_DIR}/glean/glean.swift
+    ${CMAKE_CURRENT_BINARY_DIR}/glean/generated/Metrics.swift
+)
+target_sources(iosglean PUBLIC
+    ${CMAKE_CURRENT_BINARY_DIR}/glean/gleanFFI.h
+)
+
+# We execute this as a command and not a process,
+# because different from the Glean internal stuff 
+# the VPN metrics and pings files can change constantly 
+# and we want to re-run the command for each build
+add_custom_command(
+    OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/generated/VPNMetrics.swift
+    DEPENDS ${CMAKE_SOURCE_DIR}/glean/metrics.yaml ${CMAKE_SOURCE_DIR}/glean/pings.yaml
+    COMMAND ${GLEAN_VENDORED_PATH}/glean-core/ios/sdk_generator.sh -o ${CMAKE_CURRENT_BINARY_DIR}/generated
+        ${CMAKE_SOURCE_DIR}/glean/metrics.yaml ${CMAKE_SOURCE_DIR}/glean/pings.yaml
+    # We need to rename otherwise XCode gets confused with the same name file name as the Glean internal metrics
+    COMMAND mv ${CMAKE_CURRENT_BINARY_DIR}/generated/Metrics.swift ${CMAKE_CURRENT_BINARY_DIR}/generated/VPNMetrics.swift
+)
+set_source_files_properties(
+    ${CMAKE_CURRENT_BINARY_DIR}/generated/VPNMetrics.swift
+    PROPERTIES GENERATED TRUE
+)
+target_sources(mozillavpn PRIVATE
+    ${CMAKE_CURRENT_BINARY_DIR}/generated/VPNMetrics.swift
+    platforms/ios/iosgleanbridge.swift
+    platforms/ios/iosgleanbridge.mm
+    platforms/ios/iosgleanbridge.h
+)
+
+target_link_libraries(iosglean PRIVATE vpnglean)
```

Comment:
This is basically the whole reason we are copying all of the Glean internal files and essentialy building the Glean iOS SDK from scratch. If we just used the bundled Framwork provided by the Glean team it would be linked to another glean-core binary. We want to have it linked to our nice vpnglean binary, which is also the one being consumed by C++. So everyone uses the same Glean no matter where it is being called.